### PR TITLE
OAK-9694: expose elastic stats in oak-index-stats and mbean

### PR DESCRIPTION
**Pull request by fabriziofortino**

### CODE REVIEW

**Comment by fabriziofortino** [comment]
cc @AngelaFabregues

File: oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextIndexEditorContext.java [hunk 1]
```diff
@@ -165,9 +164,8 @@ public void closeWriter() throws IOException {
       NodeBuilder status = definitionBuilder.child(IndexDefinition.STATUS_NODE);
       status.setProperty(IndexDefinition.STATUS_LAST_UPDATED, getUpdatedTime(currentTime), Type.DATE);
       status.setProperty("indexedNodes", indexedNodes);
-      if (storedIndexDefinitionEnabled() && reindex) {
-        NodeBuilder indexDefinition = definitionBuilder.child(STATUS_NODE);
-        indexDefinition.setProperty(IndexDefinition.REINDEX_COMPLETION_TIMESTAMP, ISO8601.format(currentTime), Type.DATE);
+      if (reindex) {
```
Comment (fabriziofortino) [inline oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextIndexEditorContext.java:167]: the reindex completion timestamp is exposed in the index stats. I could not find a reason why this was not set for NON stored index defs. I don't see issues in setting it in any case. Feedback is welcome @thomasmueller @nit0906 @averma21

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java [hunk 2]
```diff
@@ -71,147 +70,107 @@ public IndexPrinter(IndexInfoService indexInfoService, AsyncIndexInfoService asy
...
         }
 
         if (info.getCreationTimestamp() > 0){
-            keyValue("     Creation time           ", formatTime(info.getCreationTimestamp()), pw, json, format);
+            po.text("Creation time", formatTime(info.getCreationTimestamp()));
         }
 
         if (info.getReindexCompletionTimestamp() > 0){
-            keyValue("     Reindex completion time ", formatTime(info.getReindexCompletionTimestamp()), pw, json, format);
+            po.text("Reindex completion time ", formatTime(info.getReindexCompletionTimestamp()));
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java]: Please remove the trailing space here as well.

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java [hunk 3]
```diff
@@ -71,147 +70,107 @@ public IndexPrinter(IndexInfoService indexInfoService, AsyncIndexInfoService asy
...
         }
 
         if (info.getSizeInBytes() >= 0){
-            keyValue("    Size                     ", IOUtils.humanReadableByteCount(info.getSizeInBytes()), pw, json, format);
-            keyValue("    Size (in Bytes)          ", info.getSizeInBytes(), pw, json, format);
+            po.text("Size", IOUtils.humanReadableByteCount(info.getSizeInBytes()));
+            po.text("Size (in Bytes)", info.getSizeInBytes());
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java]: "Bytes" should be lower case in my view

**Review by thomasmueller — CHANGES_REQUESTED** [review]

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java (hunk 2 above)
Comment (fabriziofortino) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java]: Good catch! Done.

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java (hunk 3 above)
Comment (fabriziofortino) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java]: done

File: oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextIndexEditorContext.java (hunk 1 above)
Comment (thomasmueller) [reply oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextIndexEditorContext.java:167]: I don't know, but I'm quite sure @tihom88  would know; could we ask him? (I know he is on PTO, but maybe we could still ask him)

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java (hunk 2 above)
Comment (thomasmueller) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java]: (Space is still there)

**Review by thomasmueller — APPROVED** [review]

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java (hunk 2 above)
Comment (fabriziofortino) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java]: This shows the outdated code (see the label in the header). The code in the last commit does not have the space.

**Pull request merged**